### PR TITLE
Jamchri docker ci

### PR DESCRIPTION
**Pull request by Hamster-Huey**
Add an http route for DockerHub and ACR webhook receiving endpoint to kudu. Linux only. 

Please let me know what I need to do to make this endpoint "compliant"/production-worthy. 

All it does is receive a payload and update a timestamp file in the site directory.

### CODE REVIEW

File: Kudu.Services.Web/App_Start/NinjectServices.cs [hunk 1]
```diff
@@ -531,6 +531,9 @@ public static void RegisterRoutes(IKernel kernel, RouteCollection routes)
             routes.MapHttpRoute("get-masterkey", "api/functions/admin/masterkey", new { controller = "Function", action = "GetMasterKey" }, new { verb = new HttpMethodConstraint("GET") });
             routes.MapHttpRoute("delete-function", "api/functions/{name}", new { controller = "Function", action = "Delete" }, new { verb = new HttpMethodConstraint("DELETE") });
 
+            // Docker Hook Endpoint
+            routes.MapHttpRoute("docker", "docker/hook", new { controller = "Docker", action = "ReceiveHook" }, new { verb = new HttpMethodConstraint("POST") });
```
Comment (davidebbo) [inline Kudu.Services.Web/App_Start/NinjectServices.cs]: I'll let @nickwalkmsft review, but should the route only be added on Linux?

File: Kudu.Services/Docker/DockerController.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+
+namespace Kudu.Services.Docker
+{
+    public class DockerController : ApiController
+    {
+        const string TimestampDirectory = "/home/config";
```
Comment (nickwalkmsft) [inline Kudu.Services/Docker/DockerController.cs]: Does the functionality require that the timestamp file is in the mounted site directory? If so we'll want to keep a close eye on this, as a feature that may not be too far off is allowing the site drive to be mounted somewhere besides /home.

File: Kudu.Services/Docker/DockerController.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                return Request.CreateResponse(HttpStatusCode.NotFound);
+            }
+
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step("Docker.SetDockerTimestamp"))
+            {
+                string timestampPath = Path.Combine(TimestampDirectory, "dockerTimestamp.txt");
+                try
+                {
+                    File.WriteAllText(timestampPath, DateTime.UtcNow.ToString());
```
Comment (nickwalkmsft) [inline Kudu.Services/Docker/DockerController.cs]: Can't remember - will this create the directory if it doesn't exist?

File: Kudu.Services.Web/App_Start/NinjectServices.cs (hunk 1 above)
Comment (Hamster-Huey) [inline Kudu.Services.Web/App_Start/NinjectServices.cs]: For now I only see this route being useful in the Linux ecosystem as it is purely for containers/docker images.

Comment (nickwalkmsft) [inline Kudu.Services.Web/App_Start/NinjectServices.cs]: That's what I'm saying - can you completely skip the route registration here if we're on Windows?

Comment (Hamster-Huey) [inline Kudu.Services.Web/App_Start/NinjectServices.cs]: Ah i see now. your comment didn't pop up until i refreshed this page. adding the check here makes more sense for sure.

**Pull request merged**